### PR TITLE
feat(risk-assessment): NIST 800-30 weighted risk scoring engine

### PR DESCRIPTION
**Pull request by mrizzi**
## Summary

- Add weighted risk scoring engine (`scoring.rs`) that applies NIST 800-30 category weights (PT 30%, VEX 20%, SAR 20%, DAST 15%, SAST 10%, Threat Model 5%)
- Add `CategoryScore`, `OverallScore`, and `ScoringResult` model structs to the API response
- Replace simple-average scoring in `process_document` with weighted scoring; persist result to DB
- Include risk level classification (Low/Moderate/High/Very High) and missing category tracking

Implements [JIRAPLAY-1377](https://redhat.atlassian.net/browse/JIRAPLAY-1377)

## Test plan

- [x] Unit test: weighted score computation with all 6 categories (known inputs/outputs)
- [x] Unit test: score computation with missing categories (weight re-normalization)
- [x] Unit test: risk level classification boundary values
- [x] Unit test: empty categories produce score 0 / Low
- [x] Unit test: unprocessed categories excluded from scoring
- [x] `cargo clippy` — no warnings
- [x] `cargo fmt --check` — formatting correct

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by Sourcery

Introduce a NIST 800-30–based weighted risk scoring engine and expose detailed scoring information in risk assessment results.

New Features:
- Add structured scoring models (CategoryScore, OverallScore, ScoringResult) to the risk assessment API response.
- Implement a NIST 800-30–aligned weighted scoring engine that computes per-category and overall risk scores with risk level classification and missing category tracking.

Enhancements:
- Replace simple average overall scoring with the new weighted scoring engine when updating and retrieving risk assessment results.

Tests:
- Add unit tests covering weighted score calculations, missing category re-normalization, risk level thresholds, empty input handling, and exclusion of unprocessed categories from scoring.

### CODE REVIEW

File: modules/fundamental/src/risk_assessment/service/mod.rs [hunk 1]
```diff
@@ -222,10 +223,13 @@ impl RiskAssessmentService {
             });
         }
 
+        let scoring_result = scoring::compute_scoring_result(&categories);
+
         Ok(Some(RiskAssessmentResults {
             assessment_id: assessment.id.to_string(),
             overall_score: assessment.overall_score,
             categories,
+            scoring: Some(scoring_result),
```
[the remark this answers is not in the document]
Comment (mrizzi) [reply modules/fundamental/src/risk_assessment/service/mod.rs:232]: [sdlc-workflow/verify-pr] Classified as **suggestion** — this proposes avoiding redundant computation of ScoringResult across methods. This is a general code quality improvement, not documented in CONVENTIONS.md and not a demonstrated codebase convention. No sub-task created.

File: modules/fundamental/src/risk_assessment/service/mod.rs [hunk 2]
```diff
@@ -274,10 +278,14 @@ impl RiskAssessmentService {
...
-        let count = evaluation.criteria.len() as f64;
-        let overall_score = if count > 0.0 { total / count } else { 0.0 };
+        // Recompute weighted overall score across all categories
+        let results = self
+            .get_results(assessment_id, db)
+            .await?
+            .ok_or_else(|| Error::NotFound(assessment_id.to_string()))?;
+
+        let scoring_result = scoring::compute_scoring_result(&results.categories);
+        let overall_score = scoring_result.overall.score;
```
[the remark this answers is not in the document]
Comment (mrizzi) [reply modules/fundamental/src/risk_assessment/service/mod.rs:288]: [sdlc-workflow/verify-pr] Classified as **question** — this asks for clarification about whether overall_score should always mirror the scoring engine output. No code change requested. No sub-task created.

File: modules/fundamental/src/risk_assessment/service/scoring.rs [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+    ("sar", 0.20),
+    ("dast", 0.15),
+    ("sast", 0.10),
+    ("threat_model", 0.05),
+];
+
+/// All known categories for detecting missing ones.
+const ALL_CATEGORIES: &[&str] = &["pt", "vex", "sar", "dast", "sast", "threat_model"];
+
+/// NIST 800-30 risk level thresholds.
```
[the remark this answers is not in the document]
Comment (mrizzi) [reply modules/fundamental/src/risk_assessment/service/scoring.rs:18]: [sdlc-workflow/verify-pr] Classified as **suggestion** — this proposes using a macro to derive ALL_CATEGORIES from CATEGORY_WEIGHTS. Macro-based constant derivation is not documented in CONVENTIONS.md and has no established codebase pattern (0 instances of this pattern found across the codebase). No sub-task created.

File: modules/fundamental/src/risk_assessment/service/scoring.rs [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+    if pct <= 25.0 {
+        "Low"
+    } else if pct <= 50.0 {
+        "Moderate"
+    } else if pct <= 75.0 {
+        "High"
+    } else {
+        "Very High"
+    }
+}
```
[the remark this answers is not in the document]
Comment (mrizzi) [reply modules/fundamental/src/risk_assessment/service/scoring.rs:35]: [sdlc-workflow/verify-pr] Classified as **suggestion** — this proposes adding debug_assert\! and clamping for input validation in classify_risk_level. Neither debug_assert\! nor .clamp() is an established defensive programming pattern in the codebase (0 debug_assert\! in production code, 1 isolated .clamp() usage). No sub-task created.

**Comment by mrizzi** [comment]
## Verification Report for JIRAPLAY-1377 (commit `5736452`)

| Check | Result | Details |
|-------|--------|---------|
| Review Feedback | PASS | 4 threads classified (3 suggestions, 1 question). No code change requests — no sub-tasks created. |
| Root-Cause Investigation | N/A | No sub-tasks created in Steps 4 or 10. |
| Scope Containment | PASS | All 3 files match task spec (1 created, 2 modified). No out-of-scope or missing files. |
| Diff Size | PASS | 317 additions, 4 deletions across 3 files — proportionate to scoring engine scope. |
| Commit Traceability | PASS | Single commit references `JIRAPLAY-1377` in footer (`Implements JIRAPLAY-1377`). |
| Sensitive Patterns | PASS | No matches for passwords, secrets, API keys, or private keys. |
| CI Status | FAIL | 2 CI runs failed: `Migration file of version 'm0002130_add_csaf_product_id' is missing`. Pre-existing infrastructure issue with S3 migration dumps — **unrelated to PR changes**. Sourcery review and bench checks passed. |
| Acceptance Criteria | PASS | 6 of 6 criteria met (see details below). |
| Verification Commands | N/A | CI covers these checks; local execution not performed. |

### Acceptance Criteria Details

| # | Criterion | Result | Evidence |
|---|-----------|--------|----------|
| 1 | Weighted scoring correctly applies category weights | PASS | `CATEGORY_WEIGHTS` defines PT 30%, VEX 20%, SAR 20%, DAST 15%, SAST 10%, TM 5%. `compute_weighted_score` applies weighted sum with re-normalization. |
| 2 | Overall score computed as percentage (0-100%) | PASS | Score stored as 0.0–1.0 (fractional), mapped to 0–100% range in `classify_risk_level`. |
| 3 | Missing categories handled gracefully | PASS | `compute_scoring_result` identifies missing categories via `ALL_CATEGORIES` diff. `compute_weighted_score` re-normalizes weights for available categories only. `OverallScore.missing_categories` exposes the list. |
| 4 | Risk level classification maps scores to severity tiers | PASS | `classify_risk_level` maps: 0–25% → Low, 26–50% → Moderate, 51–75% → High, 76–100% → Very High. NIST 800-30 specific (not CVSS). |
| 5 | Scores persisted to database | PASS | `overall_score: Set(Some(overall_score))` at line 292 persists via `assessment_update.update(db).await?`. |
| 6 | API results endpoint returns computed scores | PASS | `get_results` computes `ScoringResult` and returns it in `RiskAssessmentResults.scoring`. |

### Review Feedback Classifications

| Thread | File | Classification | Rationale |
|--------|------|---------------|-----------|
| Avoid recomputing scoring | `service/mod.rs` L226-232 | Suggestion | General DRY improvement; no convention match. |
| overall_score consistency | `service/mod.rs` L287-288 | Question | Asks for clarification; no code change. |
| Derive ALL_CATEGORIES from CATEGORY_WEIGHTS | `service/scoring.rs` L5-18 | Suggestion | Macro-based constant derivation: 0 instances of this pattern in codebase. |
| Validate/clamp score range | `service/scoring.rs` L18-35 | Suggestion | debug_assert!/clamp: 0 instances in production code. |

### Overall: WARN

CI failures are pre-existing infrastructure issues (missing migration dump file `m0002130_add_csaf_product_id` on S3) and are **not caused by this PR**. All acceptance criteria pass. All review feedback is classified as suggestions/questions with no required code changes.

---
*This comment was AI-generated by [sdlc-workflow/verify-pr](https://github.com/mrizzi/sdlc-plugins) v0.5.5.*

File: modules/fundamental/src/risk_assessment/service/mod.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (mrizzi) [reply modules/fundamental/src/risk_assessment/service/mod.rs:232]: @sourcery-ai where is 're_evaluate' used?
